### PR TITLE
Removing redundant downloads

### PR DESCRIPTION
**Pull request by Jchaffer787**
Added check for presence of matrix_files(tmp_dir) folder and creation if not existing. Added check for accession.h5ad file in matrix_files and downloaded if not existing. Added check for spatial folder within matrix_files and creation/download if not existing. Added check for reference .csv files and downloaded if not existing. Removed the deletion of matrix_files at the end.

### CODE REVIEW

File: scripts/flattener.py [hunk 1]
```diff
@@ -1131,13 +1138,24 @@ def main(mfinal_id):
 		
 		# Add anndata to list of final raw anndatas, only for RNAseq
 		if summary_assay in ['RNA','CITE']:
-			download_file(mxr, tmp_dir)
+			# Checking for presence of mxr file and downloading if not present
+			if mxr['s3_uri'].endswith('h5'):
+				if os.path.exists(tmp_dir + '/' + mxr_acc + '.h5') == False:
+					download_file(mxr, tmp_dir)
+			if mxr['s3_uri'].endswith('h5ad'):
```
Comment (jahilton) [inline scripts/flattener.py:1145]: should be `elif...` I believe

File: scripts/flattener.py [hunk 2]
```diff
@@ -1131,13 +1138,24 @@ def main(mfinal_id):
...
 					mxr_name = '{}.h5'.format(mxr_acc)
 				elif mxr['s3_uri'].endswith('h5ad'):
 					mxr_name = '{}.h5ad'.format(mxr_acc)
-				download_directory(mfinal_obj['spatial_s3_uri'], tmp_dir)
+				# Checking for presence of spatial directory and redownloading if present
+				if os.path.exists(tmp_dir + '/spatial') == False:
+					download_directory(mfinal_obj['spatial_s3_uri'], tmp_dir)
+				else:
+					shutil.rmtree(tmp_dir + '/spatial')
+					download_directory(mfinal_obj['spatial_s3_uri'], tmp_dir)
```
Comment (jahilton) [inline scripts/flattener.py:1158]: Can consolidate these lines...
```
if os.path.exists(tmp_dir + '/spatial'):
	shutil.rmtree(tmp_dir + '/spatial')
download_directory(mfinal_obj['spatial_s3_uri'], tmp_dir)
```

**Review by jahilton — CHANGES_REQUESTED** [review]

**Review by jahilton — APPROVED** [review]
LGTM. Will now await @jychien 's review & QA

**Comment by jychien** [comment]
Looks good to me!

**Pull request merged**